service/tablet_allocator.cc

```diff
@@ -922,7 +922,30 @@ class load_balancer {
                 return bool(t.transition);
             };
 
-            auto first_non_matching_replicas = [] (tablet_replica_set r1, tablet_replica_set r2) -> std::optional<std::pair<tablet_replica, tablet_replica>> {
+            // Enable rack-aware co-location when there's more than one rack. But in practice, merge
+            // will only be able to proceed when RF=#racks, since in this configuration, the amount
+            // of cross-rack migrations will be ZERO.
+            // When RF!=#racks, it's not always possible to achieve ZERO cross-rack migrations,
+            // which is crucial for not lowering availability and/or affecting pairing consistency.
+            auto rack_aware_enabled = std::invoke([&] () {
+                auto& topology = _tm->get_topology();
+                auto [_, rs] = get_schema_and_rs(table);
+                auto dc_rf = rs->get_replication_factor(dc);
+                const auto& racks = topology.get_datacenter_rack_nodes().at(dc);
+                if (dc_rf != racks.size()) {
+                    lblogger.warn("Table {} has RF different than number of racks in DC {}, so merge can get stuck. RF={}, racks={}",
```

Comment:
Shouldn't this warning be restricted to num_racks > 1?

Reply:
> Shouldn't this warning be restricted to num_racks > 1?

Yes, we can restrict it.

Reply:
> Also, shouldn't we return false from this block?

@raphaelsc how about the above?
If I read the code correctly (over my phone), we only return `#racks > 1` without ensuring `rf == #racks`